client/cli/src/main.rs

```diff
@@ -14,11 +15,146 @@ use mint_client::utils::{
 };
 use mint_client::{Client, UserClientConfig};
 use serde::{Deserialize, Serialize};
-use serde_json::{json, to_string_pretty};
+use serde_json::{json, Value};
+use std::error::Error;
+use std::fmt::Debug;
 use std::path::PathBuf;
-use tracing::{error, info};
+use std::process::exit;
 use tracing_subscriber::EnvFilter;
 
+#[derive(Serialize)]
+#[serde(tag = "command")]
```

Comment:
Any particular reason for using an internal tag? Core-lightning for example uses external tagging. I don't have a strong opinion on it though.

Reply:
Would this help? https://serde.rs/container-attrs.html#rename_all

Reply:
If you only going to serialize it, I don't think you need a tag. The enum here is used to unify all the possible outputs, but the cli caller will know exactly which output it expects, so it doesn't need a tag.

BTW. Possibly, eventually, each variant should be it's own type, and they should all be in a separate crate, so that callers can add that crate to their dependencies and get a free type-safety. Or maybe that's just overdoing it, not sure :D

As an example of (kind of) this approach, I can give `bitcoincore-rpc` which has a `bitcoincore-rpc-json` crate with just the API definitions: https://github.com/rust-bitcoin/rust-bitcoincore-rpc/blob/master/json/src/lib.rs

Reply:
> Or maybe that's just overdoing it, not sure :D

Depends on the purpose of this crate. If we expect people to use it for operating stuff in an automated way, it might be worth it.

Also - in that case we might want to add client version into serialized reponse.

```
#[derive(Serialize)]
pub struct CliResponse {
  version: CliVersion,
  // possibly some other important metadata
  #[serde(flatten)]
  output: CliOutput,
}
```

Reply:
Since this is not a stable api used anywhere anyway, I think landing early is preferable to avoid spinning wheels and rebasing.